src/nvim/strings.c

```diff
@@ -1001,8 +1001,8 @@ int vim_vsnprintf(char *str, size_t str_m, const char *fmt, va_list ap,
             switch (length_modifier) {
               case '\0':
               case 'h': {
-                // char and short arguments are passed as int
-                arg = (tvs ? (int)tv_nr(tvs, &arg_idx) : va_arg(ap, int));
+                // char and short arguments are passed as int16_t
+                arg = (int16_t)(tvs ? tv_nr(tvs, &arg_idx) : va_arg(ap, int));
```

Comment:
👍  looks like this was caused by 079c47ed7b203b724690b627b2ad5a18f55fa2cd. But I think it needs `va_arg(ap, int16_t)`

Reply:
Do I need to update something so that `va_arg(ap, int16_t)` is possible?

Reply:
@janlazo I think `va_arg(ap, int)` is ok. 

Nice fix!